### DISCUSSION TITLE
Globally extending pass_env

### DISCUSSION THREAD

**Discussion by Cadair** [tox-dev/tox#3225] [Q&A]
Hello,

I am experimenting with nixos (opinions yet to be formed). I am using micromamba to set up environments, which is working well. However, to make micromamba happy on nixos I need to pass through the `NIX_LD` and `NIX_LD_LIBRARY_PATH` environment variables. If I do this in my projects tox.ini file like so:

```
[testenv:.pkg]
passenv = 
    NIX_LD
    NIX_LD_LIBRARY_PATH

[testenv]
passenv = 
    NIX_LD
    NIX_LD_LIBRARY_PATH
```
it works fine.

What I would like to know is if there is a way for me to configure my system to always pass these through for all my tox runs. I have tried to make a user config file:

```console
$ cat ~/.config/tox/config.ini 
[testenv:.pkg]
pass_env = 
  NIX_LD
  NIX_LD_LIBRARY_PATH

[testenv]
passenv = 
  NIX_LD
  NIX_LD_LIBRARY_PATH
```

but that didn't work.

I also tried this:
```console
$  export TOX_OVERRIDE="testenv.pass_env+=NIX_LD,NIX_LD_LIBRARY_PATH"
$  tox c -e .pkg
[testenv:.pkg]
type = Pep517VirtualEnvPackager
[...]
pass_env =
  CC
  CCSHARED
  CFLAGS
  CPPFLAGS
  CURL_CA_BUNDLE
  CXX
  FORCE_COLOR
  HOME
  LANG
  LANGUAGE
  LDFLAGS
  LD_LIBRARY_PATH
  NIX_LD
  NIX_LD_LIBRARY_PATH
```

Which looked hopeful but didn't actually pass the env through. Setting `TOX_OVERRIDE="testenv:.pkg.pass_env+=NIX_LD,NIX_LD_LIBRARY_PATH"` didn't seem to work either in reality or in the output of `tox config`.

Are there any other ways I can do this? or is any of this not working a bug? Thanks!

**Comment by gaborbernat** [comment]
The `TOX_OVERRIDE` approach is correct for this. If `tox c` shows the values in `pass_env` but they are not actually passed through at runtime, that sounds like a bug. Could you verify with a minimal reproducer — e.g. add `printenv NIX_LD` to your commands and confirm the variable is set in your shell but not reaching the tox env?

As an alternative, you could set a shell alias or wrapper script that always passes `--override testenv.pass_env+=NIX_LD,NIX_LD_LIBRARY_PATH` to tox invocations.

**Comment by gaborbernat** [comment]
Converted to issue https://github.com/tox-dev/tox/issues/3860
